src/main/cpp/src/json_parser.cuh

```diff
@@ -1470,42 +1406,46 @@ class json_parser {
    */
   __device__ cudf::size_type write_unescaped_text(char* destination)
   {
-    switch (curr_token) {
-      case json_token::VALUE_STRING:
+    switch (current_token) {
+      case json_token::VALUE_STRING: {
         // can not copy from JSON directly due to escaped chars
         // rewind the pos; parse again with copy
-        return write_string(current_token_start_pos, destination, escape_style::UNESCAPED).first;
-      case json_token::VALUE_NUMBER_INT:
-        if (number_token_len == 2 && current_token_start_pos[0] == '-' &&
-            current_token_start_pos[1] == '0') {
+        char_range_reader reader(current_range());
+        return write_string(reader, destination, escape_style::UNESCAPED);
+      }
+      case json_token::VALUE_NUMBER_INT: {
```

Comment:
I don't think all of the open/close brackets are needed, are they? Just for the cases where we're doing some initialization (VALUE_STRING, FIELD_STRING, NUMBER_FLOAT)